### PR TITLE
java implementation of problem_004 and problem_005 in project_euler

### PR DESCRIPTION
**Pull request by SnehaR26**
Added the java implementation for problem_004 of project_euler.

**Fixes issue:**

**Changes:**

### CODE REVIEW

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+    public class problem_004 {
+
+      public static void main( String[] args) {
+
+               int prod,temp,sum,rem,lar = 0,i,j;
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Add space after comma.

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+            public static void main( String[] args ) {
+
+                    int lcm = 1 , i;
+
+                    for ( i = 2; i <= 20; i++ ) {
+                              if ( lcm % i == 0 ) {
+                                  continue;
+                              }
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove this newline.

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+        public class problem_005 {
+
+            public static void main( String[] args ) {
+
+                    int lcm = 1 , i;
+
+                    for ( i = 2; i <= 20; i++ ) {
+                              if ( lcm % i == 0 ) {
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove braces

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
+        public class problem_005 {
+
+            public static void main( String[] args ) {
+
+                    int lcm = 1 , i;
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove this newline

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
+    public class problem_004 {
+
+      public static void main( String[] args) {
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove this newline.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+                            while (temp > 0) { 
+                                rem = temp % 10;
+                                sum = ( sum * 10) + rem;
+                                temp /= 10;
+                            }
+                           
+                           if (( prod == sum ) && ( prod > lar ))
+                                lar = prod;
+                  }
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove this newline

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+                                sum = ( sum * 10) + rem;
+                                temp /= 10;
+                            }
+                           
+                           if (( prod == sum ) && ( prod > lar ))
+                                lar = prod;
+                  }
+                  
+               }
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove this newline.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+                            }
+                           
+                           if (( prod == sum ) && ( prod > lar ))
+                                lar = prod;
+                  }
+                  
+               }
+               
+               System.out.println("The largest palindrome made from the product of two 3-digit numbers is " + lar);
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove this as well

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+                           if (( prod == sum ) && ( prod > lar ))
+                                lar = prod;
+                  }
+                  
+               }
+               
+               System.out.println("The largest palindrome made from the product of two 3-digit numbers is " + lar);
+               
+      }   
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove this too

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
+        public class problem_005 {
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove this newline.

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+            public static void main( String[] args ) {
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove this as well

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+                    int lcm = 1 , i;
+
+                    for ( i = 2; i <= 20; i++ ) {
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Use preincrement and remove space after and before parenthesis.

File: code/online_challenges/src/project_euler/problem_005/problem_005.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+                              if ( lcm % i == 0 ) {
+                                  continue;
+                              }
+
+                              else {
+                                  lcm++;
+                                  i = 2;
+                              }
+                      }
+
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_005/problem_005.java]: Remove this newline.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 14]
```diff
@@ -0,0 +1,30 @@
+    public class problem_004 {
+
+      public static void main( String[] args) {
+
+               int prod,temp,sum,rem,lar = 0,i,j;
+
+               for( i = 100; i <= 999; i++) {
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove space after parenthesis and use preincrement .

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+
+                  for( j = 100; j <= 999; j++) {
```
Comment (mansiag) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Same as above.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java (hunk 1 above)
Comment (mansiag) [reply code/online_challenges/src/project_euler/problem_004/problem_004.java]: Rather then declaring i and j variables here, you can declare variables in the for loop as for(int i = 100; i<= 999; ++i), same for j.

File: code/online_challenges/src/project_euler/problem_005/problem_005.java (hunk 4 above)
Comment (mansiag) [reply code/online_challenges/src/project_euler/problem_005/problem_005.java]: You can declare i in for loop as for(int i = 2; i <= 20; ++i)

**Comment by Garvit-k** [comment]
@SnehaR26  Problem 005 is already approved waiting for merge
#3367

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+                            prod = i * j;
+                            temp = prod;
+                            sum = 0;
+
+                            while (temp > 0) { 
+                                rem = temp % 10;
+                                sum = ( sum * 10) + rem;
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove extra space before `sum`.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java [hunk 17]
```diff
@@ -0,0 +1,30 @@
...
+                                temp /= 10;
+                            }
+                           
+                           if (( prod == sum ) && ( prod > lar ))
```
Comment (arnavb) [inline code/online_challenges/src/project_euler/problem_004/problem_004.java]: Remove extra spaces in parentheses.

**Comment by mansiag** [comment]
@arnavb Thanks!. Learning from you.

File: code/online_challenges/src/project_euler/problem_004/problem_004.java (hunk 15 above)
Comment (SnehaR26) [reply code/online_challenges/src/project_euler/problem_004/problem_004.java]: Thank you so much @mansiag for the suggestions. :smile:  I really appreciate it.

**Comment by SnehaR26** [comment]
@arnavb  Since one of the commits in this pull request has already been implemented, do I need to create a new PR ?

**Comment by danos7777** [comment]
I'm not sure is both _004 and_005 in project_euler(#3392) fixed

**Comment by SnehaR26** [comment]
Since one of the commits has already been implemented , I am closing the current pull request. I will open a new pull request for the java implementation of problem_004 in project euler

**Pull request closed without merging**